cmake/SetupCUDA.cmake

```diff
@@ -2,23 +2,39 @@
 # the list of CUDA libraries that we are using
 FUNCTION(SET_CUDA_LIBRARIES)
   ADD_DEFINITIONS(-DMFMG_WITH_CUDA)
+  FIND_LIBRARY(CUSPARSE cusparse PATH ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+  FIND_LIBRARY(CUSOLVER cusolver PATH ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+  MESSAGE("implicit ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}")
   SET(MFMG_CUDA_LIBRARIES
-    "cusparse"
-    "cusolver"
+    ${CUSPARSE}
+    ${CUSOLVER}
     PARENT_SCOPE
     )
```

Comment:
@Rombur It feels really wrong to have to set in the parent scope without even checking that we actually found the libraries or/and the include directories.  What was the rational to not define modules?  If we were to do so, we would be able to call `find_package_handle_standard_args()` and crash at configure time if `REQUIRED` was specified.